app/client/ui/errorPages.ts

```diff
@@ -98,6 +99,19 @@ export function createNotFoundPage(appModel: AppModel, message?: string) {
   ]);
 }
 
+export function createSigninFailedPage(appModel: AppModel, message?: string) {
+  document.title = t("Signin failed{{suffix}}", {suffix: getPageTitleSuffix(getGristConfig())});
+  return pagePanelsError(appModel, t("Signin failed{{suffix}}", {suffix: ''}), [
+    cssErrorText(message ??
+      t("Failed to login.{{separator}}Please try again or contact support.", {
+        separator: dom('br')
+    })),
+    cssButtonWrap(bigPrimaryButtonLink(t("Login again"), testId('error-primary-btn'),
+      urlState().setLinkUrl({login: 'login', params: {}}))),
```

Comment:
The function `createSignedOutPage` in this file has a similar block for the same purpose (to sign in again), but with a slightly different text, and with URL `getLoginUrl()`. If that works correctly for your case, I suggest using the same URL, and also the same text (so that existing translations work).